core/lib/tasks/exchanges.rake

```diff
@@ -20,12 +20,13 @@ namespace :exchanges do
       begin
         item_charger.charge_for_items
       rescue Spree::UnreturnedItemCharger::ChargeFailure => e
-        failure = {message: e.message, new_order: e.new_order.try(:number)}
-      rescue Exception => e
-        failure = {message: "#{e.class}: #{e.message}"}
+        failure = { message: e.message }
+      rescue => e
```

Comment:
@gmacdougall yeah but if it's a `NoMemoryError` or something like that then you're likely to make things worse by trying to call more code.  What do you think of calling the handler right away on each exception, and if there is no handler just allowing the exception to raise right then?  We might not want to make the assumption that a store will want this code to continue charging orders when unexpected exceptions are happening.